client/extensions/woocommerce/woocommerce-services/lib/initialize-labels-state/index.js

```diff
@@ -42,6 +42,35 @@ export default data => {
 	const hasOriginAddress = addressFilled( formData.origin );
 	const hasDestinationAddress = addressFilled( formData.destination );
 
+	const customsItemsData = {};
+	forEach( formData.selected_packages, ( { items } ) => {
+		items.map(
+			( {
+				product_id,
+				weight,
+				name,
+				attributes,
+				description,
+				value,
+				hs_tariff_number,
+				origin_country,
+				parent_product_id,
+			} ) => {
+				const attributesStr = attributes ? ' (' + attributes + ')' : '';
+				const defaultDescription = name.substring( name.indexOf( '-' ) + 1 ).trim() + attributesStr;
+				customsItemsData[ product_id ] = {
```

Comment:
The `product_id` here is the same for two different variations of the same product. Since the tariff numbers can be unique per product variation, should we be relying product variation ID as an index instead?

Reply:
`product_id` is actually the variation ID (if the product has variations). So, for a product `#7: T-shirt` with variations `#42: Red shirt` and `#43: Blue shirt`:
* `Red shirt` will have `product_id = 42` and `parent_product_id = 7`
* `Blue shirt` will have `product_id = 43` and `parent_product_id = 7`

For simple products, `product_id` and `parent_product_id` is the same.

A bit confusing but this "convention" seems to be all around our plugin, not sure if it's worth it to tackle it now. It's better to assume there's no such thing as a "parent product", the plugin's logic only deals with variations (or simple products).

The only case where the "parent product" matters is to render a link to the WP-Admin product page. Variations don't have their own product page, so the "parent" ID must be used.

Reply:
Hm, that's not what I'm seeing when testing. When logging `customsItemsData` in an order that has different variations of the same product, I only see one item with the product id that belongs to the 'main'/first product:

```
"47":{
  "defaultDescription":"Pencil Case (has variations) (Colour: green, Size: large)",
  "description":"Pencil Case (has variations) (Colour: green, Size: large)",
  "weight":19,
  "tariffNumber":"",
  "originCountry":"US"
}
```

Here's what the order looks like:

<img width="428" alt="screen shot 2018-07-19 at 3 26 06 pm" src="https://user-images.githubusercontent.com/11487924/42966292-a3048318-8b6a-11e8-9172-15434b61553e.png">

Here are the items list (taken from `State.extensions.woocommerce.woocommerceServices.147890486.shippingLabel.63.form.packages.selected.default_box.items`):

```
{
  items: [
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 11,
      width: 8,
      name: '#48 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: gold, Size: small'
    },
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 11,
      width: 8,
      name: '#48 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: gold, Size: small'
    },
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 15,
      width: 8,
      name: '#52 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: pink, Size: medium'
    },
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 15,
      width: 8,
      name: '#52 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: pink, Size: medium'
    },
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 15,
      width: 8,
      name: '#52 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: pink, Size: medium'
    },
    {
      height: 2,
      product_id: 47,
      length: 2,
      quantity: 1,
      weight: 19,
      width: 8,
      name: '#56 - Pencil Case (has variations)',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=47&action=edit',
      attributes: 'Colour: green, Size: large'
    },
    {
      height: 3,
      product_id: 29,
      length: 12,
      quantity: 1,
      weight: 10,
      width: 5,
      name: '3-white-candles - 3 White Candles',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=29&action=edit'
    },
    {
      height: 1.5,
      product_id: 30,
      length: 12,
      quantity: 1,
      weight: 4,
      width: 8.5,
      name: 'notebook - Notebook',
      url: 'https://valsteststoreusa201806.blog/wp-admin/post.php?post=30&action=edit'
    }
  ]
}
```

You can see that there are multiple items with `product_id = 47`but they have different attributes. Here are two examples:

> attributes: 'Colour: gold, Size: small'

> attributes: 'Colour: pink, Size: medium'

So, all the different variations have the same `product_id`.

Reply:
This was fixed by https://github.com/Automattic/woocommerce-services/pull/1382/commits/b5c39573538223d29d64d43b40068c33c99796da, thanks for hunting this down :)